pumpkin-protocol/src/server/play/s_update_sign.rs

```diff
@@ -0,0 +1,14 @@
+use pumpkin_core::math::position::WorldPosition;
+use pumpkin_macros::server_packet;
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+#[server_packet("play:sign_update")]
+pub struct SUpdateSign {
+    pub location: WorldPosition,
+    pub is_front_text: bool,
+    pub line_1: String,
+    pub line_2: String,
+    pub line_3: String,
+    pub line_4: String,
```

Comment:
Every line has the max length of 384. We currently have no way of automatically Deserializing the length. You would need to have to implement the Packet manually and use `buf.read_string_len(384)*

Reply:
If it's `384` characters per line, then why not just
```suggestion
    pub line_1: [char; 384],
    pub line_2: [char; 384],
    pub line_3: [char; 384],
    pub line_4: [char; 384],
```
?

Reply:
I forgot that the client is probably going to try and send something more than 384 characters and crash the server... :skull:

Reply:
> If it's 384 characters per line, then why not
> 
> ?

That's the wrong approche, It will be also a pain to actually get the lines.

Reply:
> > If it's 384 characters per line, then why not
> > ?
> 
> That's the wrong approche, It will be also a pain to actually get the lines.

You do save some memory by storing it as a slice though. Also, what do you mean "get the lines"? If you're using it like it's a `String` then you just use `String::from_slice` or `line_n.into()`. Although I do think that we probably shouldn't trust that the client is going to send 384 characters or below for a line.